contracts/Minter.sol

```diff
@@ -97,35 +103,68 @@ contract SPSMinter {
       unchecked {
         totalMinted = totalMinted + amount;
       }
-      token.mint(pools[i].receiver, amount);
 
-      emit Mint(pools[i].receiver, amount);
+      if (amount > 0){
+        token.mint(pools[i].receiver, amount);
+        emit Mint(pools[i].receiver, amount);
+      }
+
+      unchecked { ++i; }
     }
   }
 
   /**
    * @notice Add new pool, can be called by admin
    * @param newReceiver Address of the receiver
    * @param newAmount Amount of tokens per block
+   * @param newReductionBlocks Number of blocks between emission reduction
+   * @param newReductionBps Number of basis points to reduce emission
    */
-  function addPool(address newReceiver, uint256 newAmount) external onlyAdmin {
+  function addPool(address newReceiver, uint256 newAmount, uint256 newReductionBlocks, uint256 newReductionBps) external onlyAdmin {
     require(pools.length < poolsCap, 'SPSMinter: Pools cap reached');
     require(newAmount <= maxToPoolPerBlock, 'SPSMinter: Maximum amount per block reached');
-    pools.push(Pool(newReceiver, newAmount));
-    emit PoolAdded(newReceiver, newAmount);
+    require(newReductionBps <= 10000, "SPSMinter: newReductionBps cannot be larger than 10000");
```

Comment:
I'd recommend extracting the `10000` into a constant and using that anywhere you're currently using it. It will be easier to read and allow a quick change if you decide later you need more precision.

```solidity
uint256 constant BPS = 1e4;
```

Reply:
Updated: https://github.com/steem-monsters/sps-minter/commit/df5fbf6391dfacadb28296bb93e855cbe4458b95